tests/test_connections/test_ledger/test_ledger.py

```diff
@@ -69,30 +81,175 @@ async def test_not_hanging(self) -> None:
+        tolerance = 1
+        assert (
+            non_blocking_time + tolerance + wait_time_among_tasks < blocking_time
+        ), "`non_blocking_time + tolerance + wait_time_among_tasks` should be less than the `blocking_time`."
+
+        # create a blocking task lasting `blocking_time` secs
+        blocking_task = dummy_task_wrapper(
+            blocking_time,
+            AbciMessage(
+                AbciMessage.Performative.DUMMY, _body={"data": b"blocking_task"}  # type: ignore
+            ),
+        )
+        blocking_dummy_envelope = Envelope(
+            to="test_blocking_to",
+            sender="test_blocking_sender",
+            message=AbciMessage(AbciMessage.Performative.DUMMY),  # type: ignore
+        )
+        with mock.patch.object(
+            self.ledger_connection, "_schedule_request", return_value=blocking_task
+        ):
+            await self.ledger_connection.send(blocking_dummy_envelope)
+
+        # create a non-blocking task lasting `non_blocking_time` secs, after `wait_time_among_tasks`
+        await asyncio.sleep(wait_time_among_tasks)
+        normal_task = dummy_task_wrapper(
+            non_blocking_time,
+            AbciMessage(AbciMessage.Performative.DUMMY, _body={"data": b"normal_task"}),  # type: ignore
+        )
+        normal_dummy_envelope = Envelope(
+            to="test_normal_to",
+            sender="test_normal_sender",
+            message=AbciMessage(AbciMessage.Performative.DUMMY),  # type: ignore
+        )
+        with mock.patch.object(
+            self.ledger_connection, "_schedule_request", return_value=normal_task
+        ):
+            await self.ledger_connection.send(normal_dummy_envelope)
+
+        # the response envelopes of the ledger connection should be empty
+        assert (
+            self.ledger_connection.response_envelopes.empty()
+        ), "The response envelopes of the ledger connection should be empty."
+        # `receive()` should not be done,
+        # and multiplexer's `_receiving_loop` should be still running and have an empty `in_queue`
+        assert (
+            len(self._multiplexer.in_queue.queue) == 0
+        ), "The multiplexer's `in_queue` should not contain anything."
+
+        # sleep for `non_blocking_time + tolerance`
+        await asyncio.sleep(non_blocking_time + tolerance)
+
+        # the normal task should be finished
+        assert normal_task.done(), "Normal task should be done at this point."
+
+        # the response envelopes of the ledger connection should now contain the `normal_dummy_envelope`
+        assert (
+            not self.ledger_connection.response_envelopes.empty()
+        ), "The response envelopes of the ledger connection should not be empty."
+
+        # `receive()` should be done,
+        # and multiplexer's `_receiving_loop` should have put the `normal_dummy_envelope` in the `in_queue`
+        self._multiplexer.disconnect()
+        envelope = self._multiplexer.get(block=True)
```

Comment:
The multiplexer's `in_queue` only gets filled if we call `disconnect()`. Otherwise, the test hangs forever in the `get()`, since [this](https://github.com/valory-xyz/open-aea/blob/fd6d615eae4bd01a7dc5e045eee401b895d3f1bb/aea/multiplexer.py#L524-L526) does not return.

Does this indicate the same issue that we had with the connection in the multiplexer as well?

Reply:
Hm I think you have to place an `asyncio.sleep()` here instead of the disconnect

Reply:
Tried that without luck.

Reply:
This was an error in the test's implementation. 
We are now using the connection's `send` indirectly via the multiplexer.